resources/qml/pages/MessagingPage.qml

```diff
@@ -260,17 +263,42 @@ Page {
             property var phT: (userHasOmemo && useOmemo) ? "Omemo: " + enterMsg : enterMsg;
 
             placeholderText: phT;
-            width: parent.width - 100
+
             font {
                 family: Theme.fontFamily
                 pixelSize: Theme.fontSizeMedium
             }
 
-            height: editbox.activeFocus ? font.pixelSize * 6 : font.pixelSize * 4
-
             onTextChanged: {
                 sendButton.icon.source = getSendButtonImage()
             }
+
+           anchors.bottom: parent.bottom 
+           width: parent.width - sendButton.width - Theme.horizontalPageMargin
+        }
+        Image {
+            id: previewAttachment
+            visible: sendmsgview.attachmentPath.length > 0
+            fillMode: Image.PreserveAspectFit
+            width: parent.width /4     
+            height: parent.width /4                    
+            anchors {                                                                                                                               
+                left: parent.left; 
+                bottom: parent.bottom; bottomMargin: Theme.paddingMedium                           
+            }
+            IconButton {
+                id: removeAttachment
+                anchors {                                                                                                                               
+                    right: parent.right;   
+                    top: parent.top;                            
+                }    
+                icon.source: "image://theme/icon-splus-cancel" 
+
+                onClicked: {
+                    sendmsgview.attachmentPath = "";
+                    sendButton.icon.source = getSendButtonImage();
+                }           
+            }
```

Comment:
I like the preview very much. Maybe you can also hook the HttpFileUploader::updateStatus also to the preview? With this, one can see the progress of the upload.

Reply:
Hi, I will add the preview. For showing the progress of the upload, I need your feedback. I think it would be better to immediately store in the database a message to be sent instead of storing it once the message sent. It allows to immediately show the message in the page and to restart sending it in case of failure (through a command in the popup).  It requires to add 2 new states (sending in progress) and sending failed. Then I can also check how to add the percentage during the sending.

Reply:
I dont know if it is easy to change the sequence. If you want to store the message before sending, you must at least request the put and get URLs from the server. If this is done, the message could be sent to the client (and with this stored in the db). But the receiving client won't be able to download the image, because it is not on the server yet. There is currently no interface to only store the message in db without actually sending it. 

An easier way would be to use dedicated Qt signals and attach the signals to the qml. Like httpUploadStarted and httpUploadFinished. That signals could be used to activate a qmlLabel view and in this display the upload status.

Any other ideas? Discussion about a good solution is welcome ;-).

Reply:
Another option is of course to accept this PR now as it is and think about the http upload status later on. What do you prefer?

Reply:
Yes. I prefer this way. I don't think I will be able to move forward in the coming days